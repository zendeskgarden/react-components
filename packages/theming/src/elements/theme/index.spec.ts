/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import DEFAULT_THEME from '.';

describe('DEFAULT_THEME', () => {
  it('matches snapshot', () => {
    expect({
      ...DEFAULT_THEME,
      shadows: {
        sm: DEFAULT_THEME.shadows.sm('black'),
        md: DEFAULT_THEME.shadows.md('black'),
        lg: DEFAULT_THEME.shadows.lg('0', '0', 'black')
      }
    }).toMatchSnapshot();
  });

  it('does not include `PALETTE.product`', () => {
    expect(DEFAULT_THEME.palette.product).toBeUndefined();
  });
});
