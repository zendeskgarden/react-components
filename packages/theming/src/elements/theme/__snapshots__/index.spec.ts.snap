// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DEFAULT_THEME matches snapshot 1`] = `
{
  "borderRadii": {
    "md": "4px",
    "sm": "2px",
  },
  "borderStyles": {
    "solid": "solid",
  },
  "borderWidths": {
    "md": "3px",
    "sm": "1px",
  },
  "borders": {
    "md": "3px solid",
    "sm": "1px solid",
  },
  "breakpoints": {
    "lg": "992px",
    "md": "768px",
    "sm": "576px",
    "xl": "1200px",
    "xs": "0px",
  },
  "colors": {
    "base": "light",
    "chromeHue": "kale",
    "dangerHue": "red",
    "neutralHue": "grey",
    "primaryHue": "blue",
    "successHue": "green",
    "variables": {
      "dark": {
        "background": {
          "danger": "dangerHue.1000",
          "dangerEmphasis": "dangerHue.600",
          "default": "neutralHue.1100",
          "disabled": "rgba(neutralHue.500, 100)",
          "emphasis": "neutralHue.600",
          "primary": "primaryHue.900",
          "primaryEmphasis": "primaryHue.600",
          "raised": "neutralHue.1000",
          "recessed": "neutralHue.1200",
          "subtle": "neutralHue.1000",
          "success": "successHue.1000",
          "successEmphasis": "successHue.600",
          "warning": "warningHue.1000",
          "warningEmphasis": "warningHue.600",
        },
        "border": {
          "danger": "dangerHue.800",
          "dangerEmphasis": "dangerHue.600",
          "default": "neutralHue.800",
          "disabled": "neutralHue.800",
          "emphasis": "neutralHue.600",
          "primaryEmphasis": "primaryHue.600",
          "subtle": "neutralHue.900",
          "success": "successHue.800",
          "successEmphasis": "successHue.600",
          "warning": "warningHue.800",
          "warningEmphasis": "warningHue.600",
        },
        "foreground": {
          "danger": "dangerHue.400",
          "dangerEmphasis": "dangerHue.300",
          "default": "neutralHue.300",
          "disabled": "neutralHue.700",
          "onEmphasis": "neutralHue.1100",
          "primary": "primaryHue.600",
          "subtle": "neutralHue.500",
          "success": "successHue.400",
          "successEmphasis": "successHue.300",
          "warning": "warningHue.400",
          "warningEmphasis": "warningHue.300",
        },
      },
      "light": {
        "background": {
          "danger": "dangerHue.100",
          "dangerEmphasis": "dangerHue.700",
          "default": "palette.white",
          "disabled": "rgba(neutralHue.700, 100)",
          "emphasis": "neutralHue.700",
          "primary": "primaryHue.100",
          "primaryEmphasis": "primaryHue.700",
          "raised": "palette.white",
          "recessed": "neutralHue.100",
          "subtle": "neutralHue.100",
          "success": "successHue.100",
          "successEmphasis": "successHue.700",
          "warning": "warningHue.100",
          "warningEmphasis": "warningHue.700",
        },
        "border": {
          "danger": "dangerHue.300",
          "dangerEmphasis": "dangerHue.700",
          "default": "neutralHue.300",
          "disabled": "neutralHue.300",
          "emphasis": "neutralHue.600",
          "primaryEmphasis": "primaryHue.700",
          "subtle": "neutralHue.200",
          "success": "successHue.300",
          "successEmphasis": "successHue.700",
          "warning": "warningHue.300",
          "warningEmphasis": "warningHue.700",
        },
        "foreground": {
          "danger": "dangerHue.700",
          "dangerEmphasis": "dangerHue.900",
          "default": "neutralHue.900",
          "disabled": "neutralHue.600",
          "onEmphasis": "palette.white",
          "primary": "primaryHue.700",
          "subtle": "neutralHue.700",
          "success": "successHue.700",
          "successEmphasis": "successHue.900",
          "warning": "warningHue.700",
          "warningEmphasis": "warningHue.900",
        },
      },
    },
    "warningHue": "yellow",
  },
  "components": {},
  "fontSizes": {
    "lg": "18px",
    "md": "14px",
    "sm": "12px",
    "xl": "22px",
    "xs": "10px",
    "xxl": "26px",
    "xxxl": "36px",
  },
  "fontWeights": {
    "black": 900,
    "bold": 700,
    "extrabold": 800,
    "extralight": 200,
    "light": 300,
    "medium": 500,
    "regular": 400,
    "semibold": 600,
    "thin": 100,
  },
  "fonts": {
    "mono": "SFMono-Regular,Consolas,"Liberation Mono",Menlo,Courier,monospace",
    "system": "system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",Arial,sans-serif",
  },
  "iconSizes": {
    "lg": "26px",
    "md": "16px",
    "sm": "12px",
  },
  "lineHeights": {
    "lg": "24px",
    "md": "20px",
    "sm": "16px",
    "xl": "28px",
    "xxl": "32px",
    "xxxl": "44px",
  },
  "opacity": {
    "100": 0.08,
    "1000": 0.8,
    "1100": 0.88,
    "1200": 0.96,
    "200": 0.16,
    "300": 0.24,
    "400": 0.32,
    "500": 0.4,
    "600": 0.48,
    "700": 0.56,
    "800": 0.64,
    "900": 0.72,
  },
  "palette": {
    "azure": {
      "100": "#eff7fe",
      "1000": "#122238",
      "1100": "#0e1a2a",
      "1200": "#070d14",
      "200": "#d9ecfc",
      "300": "#c4e0fa",
      "400": "#82bcf4",
      "500": "#4b9fee",
      "600": "#3191ea",
      "700": "#2770c3",
      "800": "#23446b",
      "900": "#1a3250",
    },
    "black": "#000",
    "blue": {
      "100": "#edf7ff",
      "1000": "#0a2338",
      "1100": "#061420",
      "1200": "#040d15",
      "200": "#ddecf8",
      "300": "#cce0f1",
      "400": "#93bcdc",
      "500": "#66a0cd",
      "600": "#2694d6",
      "700": "#1f73b7",
      "800": "#13456d",
      "900": "#0f3655",
    },
    "crimson": {
      "100": "#fbf3f2",
      "1000": "#440e09",
      "1100": "#340b07",
      "1200": "#1c0604",
      "200": "#f7e7e4",
      "300": "#f1d7d2",
      "400": "#e2aaa0",
      "500": "#d58678",
      "600": "#cf7464",
      "700": "#be4938",
      "800": "#811b12",
      "900": "#61140d",
    },
    "fuschia": {
      "100": "#fbf3f8",
      "1000": "#3f0939",
      "1100": "#31072c",
      "1200": "#1b0418",
      "200": "#f7e6f1",
      "300": "#f2d5e7",
      "400": "#e3a4cc",
      "500": "#d77db7",
      "600": "#d16aac",
      "700": "#b34496",
      "800": "#78116c",
      "900": "#5a0d51",
    },
    "green": {
      "100": "#eef8f4",
      "1000": "#0c261c",
      "1100": "#0a1511",
      "1200": "#080d0c",
      "200": "#daeee6",
      "300": "#cae3d9",
      "400": "#94c1b0",
      "500": "#4eab89",
      "600": "#26a178",
      "700": "#037f52",
      "800": "#104b35",
      "900": "#0b3b29",
    },
    "grey": {
      "100": "#f8f9f9",
      "1000": "#1c2227",
      "1100": "#151a1e",
      "1200": "#0a0d0e",
      "200": "#e8eaec",
      "300": "#d8dcde",
      "400": "#b0b8be",
      "500": "#919ca5",
      "600": "#848f99",
      "700": "#5c6970",
      "800": "#39434b",
      "900": "#293239",
    },
    "kale": {
      "100": "#ecf9f9",
      "1000": "#03252a",
      "1100": "#061517",
      "1200": "#060e0e",
      "200": "#daeded",
      "300": "#cbe2e1",
      "400": "#97bfbf",
      "500": "#6ba4a5",
      "600": "#4a9999",
      "700": "#40787a",
      "800": "#16494f",
      "900": "#063940",
    },
    "lemon": {
      "100": "#fff7d4",
      "1000": "#2b2000",
      "1100": "#221800",
      "1200": "#110c00",
      "200": "#ffea97",
      "300": "#ffdc4f",
      "400": "#efab00",
      "500": "#c79100",
      "600": "#b68500",
      "700": "#8f6900",
      "800": "#563e00",
      "900": "#3f2e00",
    },
    "lime": {
      "100": "#ecfae7",
      "1000": "#16250e",
      "1100": "#111d0a",
      "1200": "#090e05",
      "200": "#d1f3c7",
      "300": "#b3eda3",
      "400": "#4fd12b",
      "500": "#45b025",
      "600": "#509f2d",
      "700": "#3d7e19",
      "800": "#2c491b",
      "900": "#203614",
    },
    "mint": {
      "100": "#d6ffeb",
      "1000": "#0e261a",
      "1100": "#0b1d14",
      "1200": "#050e0a",
      "200": "#9affce",
      "300": "#0afe89",
      "400": "#00d26d",
      "500": "#01b15c",
      "600": "#16a260",
      "700": "#2d7e55",
      "800": "#1b4b33",
      "900": "#143726",
    },
    "orange": {
      "100": "#fdf3ed",
      "1000": "#361a0c",
      "1100": "#291409",
      "1200": "#150a04",
      "200": "#fae7d8",
      "300": "#f7d7be",
      "400": "#eda875",
      "500": "#e58035",
      "600": "#d57428",
      "700": "#af5626",
      "800": "#693317",
      "900": "#4d2711",
    },
    "pink": {
      "100": "#fcf3f4",
      "1000": "#3c141f",
      "1100": "#2e0f18",
      "1200": "#17080c",
      "200": "#f7e5e8",
      "300": "#f3d6dc",
      "400": "#e5a6b4",
      "500": "#d98193",
      "600": "#d96b81",
      "700": "#d62054",
      "800": "#75263d",
      "900": "#561d2e",
    },
    "purple": {
      "100": "#f9f3fb",
      "1000": "#2e1150",
      "1100": "#230d3f",
      "1200": "#120720",
      "200": "#f2e7f6",
      "300": "#e9d8f1",
      "400": "#d0a9e0",
      "500": "#bb86d3",
      "600": "#b276cd",
      "700": "#9256b1",
      "800": "#58209a",
      "900": "#411973",
    },
    "red": {
      "100": "#fff2f3",
      "1000": "#3d1418",
      "1100": "#1d1011",
      "1200": "#100b0c",
      "200": "#fee3e5",
      "300": "#f5d5d8",
      "400": "#f2a1a8",
      "500": "#ea7881",
      "600": "#eb5c69",
      "700": "#cd3642",
      "800": "#7e1d25",
      "900": "#671219",
    },
    "royal": {
      "100": "#f4f5fc",
      "1000": "#0d1a5a",
      "1100": "#0a1445",
      "1200": "#050a25",
      "200": "#e7e9f8",
      "300": "#d8dcf4",
      "400": "#acb4e7",
      "500": "#8a96dd",
      "600": "#7a88d9",
      "700": "#4c67d3",
      "800": "#1833ab",
      "900": "#122680",
    },
    "teal": {
      "100": "#d4fefa",
      "1000": "#122423",
      "1100": "#0e1c1a",
      "1200": "#070d0d",
      "200": "#88fdf1",
      "300": "#0bf8e1",
      "400": "#03cdb8",
      "500": "#02ad9c",
      "600": "#2a9d8f",
      "700": "#367a74",
      "800": "#254846",
      "900": "#1b3534",
    },
    "white": "#fff",
    "yellow": {
      "100": "#fff3e4",
      "1000": "#2d1e15",
      "1100": "#18120f",
      "1200": "#0e0c0b",
      "200": "#ffe6cb",
      "300": "#fed6a9",
      "400": "#fca347",
      "500": "#e38215",
      "600": "#d67305",
      "700": "#ac5918",
      "800": "#673515",
      "900": "#4c2c17",
    },
  },
  "rtl": false,
  "shadowWidths": {
    "md": "3px",
    "sm": "2px",
    "xs": "1px",
  },
  "shadows": {
    "lg": "0 0 0 0 black",
    "md": "0 0 0 3px black",
    "sm": "0 0 0 2px black",
    "xs": "0 0 0 1px black",
  },
  "space": {
    "base": 4,
    "lg": "32px",
    "md": "20px",
    "sm": "12px",
    "xl": "40px",
    "xs": "8px",
    "xxl": "48px",
    "xxs": "4px",
  },
}
`;
